### PR TITLE
fix ATS + add Euro Truck Simulator 2

### PR DESCRIPTION
**Pull request by QuintenQVD0**
# Description

add Euro Truck Simulator 2 egg

closes: #1997
## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?

### CODE REVIEW

File: game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/egg-euro-truck-simulator2-dedicated-server.json [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "query port",
+            "description": "",
+            "env_variable": "QUERY_PORT",
+            "default_value": "27016",
+            "user_viewable": true,
+            "user_editable": false,
+            "rules": "required|string|between:27015,27020",
```
Comment (kj4lxc) [inline game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/egg-euro-truck-simulator2-dedicated-server.json:80]: Just an FYI just ran into this on my own setup, that will require a string between 27015 and 27020 characters long..... not between those two numerical values.

Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/egg-euro-truck-simulator2-dedicated-server.json:80]: haha I Will fix and also fix it for your egg :)

Comment (kj4lxc) [reply game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/egg-euro-truck-simulator2-dedicated-server.json:80]: THANK YOU!!!!

**Comment by gOOvER** [comment]
`required|string|between:27015,27020`

Is there any need that only 5 Ports will work?

**Comment by QuintenQVD0** [comment]
> `required|string|between:27015,27020`
> 
> Is there any need that only 5 Ports will work?

the between in regex is removed, but it there readme that is what the tell

File: game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+- server_packages.sii
+- server_packages.dat
+
+
+# Copy to Server
+
+With your newly set up files its time to copy them over to the Pterodactyl Panel.  Simply upload them through either your preferred SCP program or the file browser in the server panel to the .local/share/Euro Truck Simulator 2/ folder.
+
+On the panel select start server and the server will then start up and create a default server_config.sii file.  Proceed to the .local/share/Euro Truck Simulator 2/folder.
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/README.md:39]: there is no such a dir: `.local/share/Euro Truck Simulator 2/`

there is only :
chrome_Tp8F2uFhP3
chrome_XvuvsTvegl

Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/README.md:39]: it should be generate on install script

File: game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/egg-euro-truck-simulator2-dedicated-server.json [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
...d_linux.tar.gz\r\ntar -xzvf steamcmd.tar.gz -C \/mnt\/server\/steamcmd\r\nmkdir -p \/mnt\/server\/steamapps # Fix steamcmd disk write error when this folder is missing\r\ncd \/mnt\/server\/steamcmd\r\n\r\n# SteamCMD fails otherwise for some reason, even running as root.\r\n# This is changed at the end of the install process anyways.\r\nchown -R root:root \/mnt\r\nexport HOME=\/mnt\/server\r\n\r\n## install game using steamcmd\r\n.\/steamcmd.sh +force_install_dir \/mnt\/server +login ${STEAM_USER} ${STEAM_PASS} ${STEAM_AUTH} +app_update ${SRCDS_APPID} validate +quit\r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\nmkdir -p \/mnt\/server\/.local\/share\/Euro Truck Simulator 2\/\r\n\r\necho -e '\\033[1mUpload your config and pack file first before starting the server!\\033[0m'\r\n\r\n\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/egg-euro-truck-simulator2-dedicated-server.json:27]: Spaces are not possible, when you creating a Dir

Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/egg-euro-truck-simulator2-dedicated-server.json:27]: I remember there being "" around that

Comment (gOOvER) [reply game_eggs/steamcmd_servers/truck-simulator/euro-truck-simulator2/egg-euro-truck-simulator2-dedicated-server.json:27]: this could work

**Comment by gOOvER** [comment]
Next problem i have, the server_config.sii is empty and ends up in an error.

**Comment by QuintenQVD0** [comment]
> 

The readme tells you that you need to have a export pack and a config file that was generated by your game to be upload first before even starting the server

**Comment by QuintenQVD0** [comment]
as the config is unique to the exported package

**Comment by gOOvER** [comment]
chrome_9KCw6lTT6M


both are exported and uploaded

**Comment by QuintenQVD0** [comment]
> chrome_9KCw6lTT6M
> 
> both are exported and uploaded

your config is clearly empty

**Comment by QuintenQVD0** [comment]
> deleted all, uploaded complete new and servercfg is still 0

Strange as I also uploaded mine (it is not mine, I am lending it from someone) and it works fine.

**Comment by QuintenQVD0** [comment]
> deleted all, uploaded complete new and servercfg is still 0

Is this config and pack generate in game or in the "garage"?

**Comment by gOOvER** [comment]
> 

like mentoned in Readme; created a Convoi and after started Convoi i exported the server

**Comment by QuintenQVD0** [comment]
> server

an example of the contence of the file is givven, it is just 1 line that is unique

**Comment by gOOvER** [comment]
you mean this:

**Comment by kj4lxc** [comment]
Try removing the parser config from the egg and starting the server with your uploads.  If it comes online then the parser is clearing the file.... I had it happen once with my dev for ATS, but it only did it when the server tried to create the server_config.sii when one wasnt there.

**Comment by gOOvER** [comment]
same error, when i export the serverfiles, when im sitting in my Truck

**Comment by QuintenQVD0** [comment]
> you mean this:
> 
> ```
> SiiNunit
> {
> server_packages_info : _nameless.1d5.246c.0af0 {
>  version: 1
>  dlc_essential_list: 0
>  dlc_non_essential_list: 6
>  dlc_non_essential_list[0]: 2583912098162477
>  dlc_non_essential_list[1]: 92094914797
>  dlc_non_essential_list[2]: 92649725589
>  dlc_non_essential_list[3]: 59316381
>  dlc_non_essential_list[4]: 63651269
>  dlc_non_essential_list[5]: 2152080662829
>  mod_list: 0
>  map_name: "/map/europe.mbd"
>  map_dimensions: (-16384000, -16384000, 20480000, 20480000)
>  roads_data_file_name: "/home/server_packages.dat"
>  time_stamp: 1670765542
> }
> 
> }
> ```

that is server_packages.sii not the config

**Comment by gOOvER** [comment]
> Try removing the parser config from the egg and starting the server with your uploads. If it comes online then the parser is clearing the file.... I had it happen once with my dev for ATS, but it only did it when the server tried to create the server_config.sii when one wasn't there. The parser just may not be feeasable with this particular environment the way they do their files, at least until they change it and flesh it out more.

you are right; the parser clear the config file

After deleting the parsing from egg, file hat 682 byte.

So parsing is broken

**Comment by gOOvER** [comment]
> > you mean this:
> > ```
> > SiiNunit
> > {
> > server_packages_info : _nameless.1d5.246c.0af0 {
> >  version: 1
> >  dlc_essential_list: 0
> >  dlc_non_essential_list: 6
> >  dlc_non_essential_list[0]: 2583912098162477
> >  dlc_non_essential_list[1]: 92094914797
> >  dlc_non_essential_list[2]: 92649725589
> >  dlc_non_essential_list[3]: 59316381
> >  dlc_non_essential_list[4]: 63651269
> >  dlc_non_essential_list[5]: 2152080662829
> >  mod_list: 0
> >  map_name: "/map/europe.mbd"
> >  map_dimensions: (-16384000, -16384000, 20480000, 20480000)
> >  roads_data_file_name: "/home/server_packages.dat"
> >  time_stamp: 1670765542
> > }
> > 
> > }
> > ```
> 
> that is server_packages.sii not the config

you cant parse this with file parser

**Comment by gOOvER** [comment]
> > you mean this:
> > ```
> > SiiNunit
> > {
> > server_packages_info : _nameless.1d5.246c.0af0 {
> >  version: 1
> >  dlc_essential_list: 0
> >  dlc_non_essential_list: 6
> >  dlc_non_essential_list[0]: 2583912098162477
> >  dlc_non_essential_list[1]: 92094914797
> >  dlc_non_essential_list[2]: 92649725589
> >  dlc_non_essential_list[3]: 59316381
> >  dlc_non_essential_list[4]: 63651269
> >  dlc_non_essential_list[5]: 2152080662829
> >  mod_list: 0
> >  map_name: "/map/europe.mbd"
> >  map_dimensions: (-16384000, -16384000, 20480000, 20480000)
> >  roads_data_file_name: "/home/server_packages.dat"
> >  time_stamp: 1670765542
> > }
> > 
> > }
> > ```
> 
> that is server_packages.sii not the config

As i said many times; config is 0.

**Comment by QuintenQVD0** [comment]
> > Try removing the parser config from the egg and starting the server with your uploads. If it comes online then the parser is clearing the file.... I had it happen once with my dev for ATS, but it only did it when the server tried to create the server_config.sii when one wasn't there. The parser just may not be feeasable with this particular environment the way they do their files, at least until they change it and flesh it out more.
> 
> you are right; the parser clear the config file
> 
> After deleting the parsing from egg, file have 682 byte.
> 
> So parsing is broken

The parser parses the config, not the server_packages

**Comment by QuintenQVD0** [comment]
> @QuintenQVD0
> 
> ```
> SiiNunit
> {
> server_config : _nameless.447.b130 {
>  lobby_name: ""
>  description: ""
>  welcome_message: ""
>  password: ""
>  max_players: 8
>  max_vehicles_total: 100
>  max_ai_vehicles_player: 50
>  max_ai_vehicles_player_spawn: 30
>  connection_virtual_port: 100
>  query_virtual_port: 101
>  connection_dedicated_port: 27015
>  query_dedicated_port: 27016
>  server_logon_token: ""
>  player_damage: true
>  traffic: true
>  hide_in_company: false
>  hide_colliding: true
>  force_speed_limiter: false
>  mods_optioning: false
>  service_no_collision: false
>  in_menu_ghosting: false
>  name_tags: true
>  friends_only: false
>  show_server: true
>  moderator_list: 0
> }
> 
> }
> ```
> 
> You cant parse this with file parser. Thats the problem

It worked fine for me

**Comment by gOOvER** [comment]
> > > Try removing the parser config from the egg and starting the server with your uploads. If it comes online then the parser is clearing the file.... I had it happen once with my dev for ATS, but it only did it when the server tried to create the server_config.sii when one wasn't there. The parser just may not be feeasable with this particular environment the way they do their files, at least until they change it and flesh it out more.
> > 
> > 
> > you are right; the parser clear the config file
> > After deleting the parsing from egg, file have 682 byte.
> > So parsing is broken
> 
> The parser parses the config, not the server_packages

Yor parsing DELETE the Serverconfig !!

**Comment by QuintenQVD0** [comment]
> > > > > Try removing the parser config from the egg and starting the server with your uploads. If it comes online then the parser is clearing the file.... I had it happen once with my dev for ATS, but it only did it when the server tried to create the server_config.sii when one wasn't there. The parser just may not be feeasable with this particular environment the way they do their files, at least until they change it and flesh it out more.
> > > > 
> > > > 
> > > > you are right; the parser clear the config file
> > > > After deleting the parsing from egg, file have 682 byte.
> > > > So parsing is broken
> > > 
> > > 
> > > The parser parses the config, not the server_packages
> > 
> > 
> > Yor parsing DELETE the Serverconfig !!
> 
> I get that, I am reinstalling to test rn

wtf, it was working like 10 min ago...

**Comment by QuintenQVD0** [comment]
> Problem is, that the file is generated after parsing. So dl first from somewhere the config file to the dir

You thing if I curl a example config there in the install script it will work?

**Comment by kj4lxc** [comment]
> > > > > > Try removing the parser config from the egg and starting the server with your uploads. If it comes online then the parser is clearing the file.... I had it happen once with my dev for ATS, but it only did it when the server tried to create the server_config.sii when one wasn't there. The parser just may not be feeasable with this particular environment the way they do their files, at least until they change it and flesh it out more.
> > > > > 
> > > > > 
> > > > > you are right; the parser clear the config file
> > > > > After deleting the parsing from egg, file have 682 byte.
> > > > > So parsing is broken
> > > > 
> > > > 
> > > > The parser parses the config, not the server_packages
> > > 
> > > 
> > > Yor parsing DELETE the Serverconfig !!
> > 
> > 
> > I get that, I am reinstalling to test rn
> 
> wtf, it was working like 10 min ago...

Yeah im re testing mine and will also do another pull to fix my egg for ATS

**Comment by gOOvER** [comment]
chrome_XziYBSCcok

After deleting the parsing from the egg

**Comment by QuintenQVD0** [comment]
> chrome_XziYBSCcok
> 
> After deleting the parsing from the egg

then the will have to set everything manual ...

**Comment by QuintenQVD0** [comment]
> > > > > > > Try removing the parser config from the egg and starting the server with your uploads. If it comes online then the parser is clearing the file.... I had it happen once with my dev for ATS, but it only did it when the server tried to create the server_config.sii when one wasn't there. The parser just may not be feeasable with this particular environment the way they do their files, at least until they change it and flesh it out more.
> > > > > > 
> > > > > > 
> > > > > > you are right; the parser clear the config file
> > > > > > After deleting the parsing from egg, file have 682 byte.
> > > > > > So parsing is broken
> > > > > 
> > > > > 
> > > > > The parser parses the config, not the server_packages
> > > > 
> > > > 
> > > > Yor parsing DELETE the Serverconfig !!
> > > 
> > > 
> > > I get that, I am reinstalling to test rn
> > 
> > 
> > wtf, it was working like 10 min ago...
> 
> Yeah im re testing mine and will also do another pull to fix my egg for ATS

I will just do it here, I already made here like 5 changes to your egg

**Comment by kj4lxc** [comment]
Okay, How do you want to handle the needed changes to the README?  I can write it up real quick and add it as a comment here so you can fix both as well?

**Comment by kj4lxc** [comment]
> Okay, How do you want to handle the needed changes to the README? I can write it up real quick and add it as a comment here so you can fix both as well?

Actually the readme is already written for both of ours for the correct way.  I never changed it when we added in the vars.  Its already worded correctly so you are good to go for both of us!

**Comment by QuintenQVD0** [comment]
> > Okay, How do you want to handle the needed changes to the README? I can write it up real quick and add it as a comment here so you can fix both as well?
> 
> Actually the readme is already written for both of ours for the correct way. I never changed it when we added in the vars. Its already worded correctly so you are good to go for both of us!

Yes indeed, the will just have to manually change the port in the config

**Comment by QuintenQVD0** [comment]
@gOOvER Can you test the egg on how it is RN?

**Comment by gOOvER** [comment]
> @gOOvER Can you test the egg on how it is RN?

without parser, config is generatd and server starts without any error

**Comment by QuintenQVD0** [comment]
> > @gOOvER Can you test the egg on how it is RN?
> 
> without parser, config is generatd and server starts without any error

So now eggs are fine, just the have to manually change the server ports. Now waiting on a staff to approve this

**Comment by kj4lxc** [comment]
> > > @gOOvER Can you test the egg on how it is RN?
> > 
> > 
> > without parser, config is generatd and server starts without any error
> 
> So now eggs are fine, just the have to manually change the server ports. Now waiting on a staff to approve this

Excellent!  Had a blast getting to work with you on this and my previous PR.  Glad to have had the experience.

**Comment by gOOvER** [comment]
MB you can add to the readme, that the token must be created with App_ID 

`227300`

Else server crash on start

**Comment by QuintenQVD0** [comment]
> MB you can add to the readme, that the token must be created with App_ID
> 
> `227300`
> 
> Else server crash on start

realy, is that realy needed or will the figure it out

**Comment by gOOvER** [comment]
> > MB you can add to the readme, that the token must be created with App_ID
> > `227300`
> > Else server crash on start
> 
> realy, is that realy needed or will the figure it out

didn't test it without the tooken, but i think it's needed, when you can set it in the config

**Comment by kj4lxc** [comment]
> > > MB you can add to the readme, that the token must be created with App_ID
> > > `227300`
> > > Else server crash on start
> > 
> > 
> > realy, is that realy needed or will the figure it out
> 
> didn't test it without the tooken, but i think it's needed, when you can set it in the config

Token is not required, only if you want a "persistent" server instance to be registered.  I can update the readme with the needed infos and post it

**Comment by QuintenQVD0** [comment]
> > > > MB you can add to the readme, that the token must be created with App_ID
> > > > `227300`
> > > > Else server crash on start
> > > 
> > > 
> > > realy, is that realy needed or will the figure it out
> > 
> > 
> > didn't test it without the tooken, but i think it's needed, when you can set it in the config
> 
> Token is not required, only if you want a "persistent" server instance to be registered. I can update the readme with the needed infos and post it

Token indeed is not needed, Just send the changes here, I just add them to both eggs then this is just 1 big PR

**Comment by kj4lxc** [comment]
[README.md](https://github.com/parkervcp/eggs/files/10202440/README.md)

This should be the updated readme for ATS just grab the blurb at the bottom and past to yours, i put both codes in for ease.

**Review by gOOvER — APPROVED** [review]
after the changes the server start and i can join:

```
[MP] =========================================================================
[MP] Session automated start
[MP] -------------------------------------------------------------------------
[MP] DLC - eu_balkane
[MP] DLC - eu_balt
[MP] DLC - eu_east
[MP] DLC - eu_fr
[MP] DLC - eu_it
[MP] DLC - eu_north
[MP] -------------------------------------------------------------------------
[MP] Server init
dlopen failed trying to load:
steamclient.so
with error:
steamclient.so: cannot open shared object file: No such file or directory
[S_API] SteamAPI_Init(): Loaded '/home/container/.steam/sdk64/steamclient.so' OK.  (First tried local 'steamclient.so')
CAppInfoCacheReadFromDiskThread took 2 milliseconds to initialize
Setting breakpad minidump AppID = 227300
[MP] LogOn to persistent server account.
[MP] Steam connected
[MP] Settings init
[MP] Init steam game server params
[MP] Initiate server-client.
[MP] Session started.
[MP] Server created.
[MP] =========================================================================
[MP] 
[MP] Session search id: 85568392925782108/101
[MP] Session name: goover.de - Austrian Drivers
[MP] Session description: 
[MP] Maximum number of players: 8
[MP] Connection virtual ports: 100 / 101
[MP] Connection server ports: 27020 / 27021
[MP] Friends only: False
[MP] 
[MP] Damage from players: True
[MP] Traffic enabled: True
[MP] Hide player in company: False
[MP] Hide colliding players: True
[MP] Force speed limiter: False
[MP] Show name tags: True
[MP] Timezones: False
[MP] Disable collisions in service: False
[MP] Timezones: False
[MP] Mods optioning: False
[MP] 
[MP] Game version: 1.46s
[MP] Map name: /map/europe.mbd
[MP] Password required: Yes
[MP] Modded session: No
[MP] 
[MP] =========================================================================
[MP] Session running.
[MP] State: running;  Time: 2329;  Players: 0
container@pterodactyl~ Server marked as running...
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
[MP] State: running;  Time: 182334;  Players: 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
[MP] State: running;  Time: 362335;  Players: 0
[MP] [Chat] gOOvER|OldNoobTV connected
[MP] gOOvER|OldNoobTV connected, client_id = 10 
[MP] [Chat] gOOvER|OldNoobTV disconnected
[MP] gOOvER|OldNoobTV disconnected, client_id = 10 
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
[MP] State: running;  Time: 542336;  Players: 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
src/steamnetworkingsockets/clientlib/steamnetworkingsockets_sdr_common.h (564) : m_pServer->m_nReplyTimeoutsSinceLastRecv == 0
[MP] State: running;  Time: 722337;  Players: 0
```

**Comment by nousername363** [comment]
cant use with account linked with aunthenticator on phone cant enter the code

**Comment by gOOvER** [comment]
> cant use with account linked with aunthenticator on phone cant enter the code

you dont need to login to dowload the server

**Comment by QuintenQVD0** [comment]
@kj4lxc @gOOvER I added back the file parser as the install script no download an example config. where they just have to change 1 line. Have a look at https://github.com/parkervcp/eggs/pull/2002/commits/5e4518ed0457cc698ca45fbec3881688a6d33a70 It wont work yet as the example config is not yet merged to the master branch.

**Comment by kj4lxc** [comment]
First startup will make a config for them to modify.  If the parser is in it will clear it out and the server won't recover

**Comment by kj4lxc** [comment]
> @kj4lxc @gOOvER I added back the file parser as the install script no download an example config. where they just have to change 1 line. Have a look at https://github.com/parkervcp/eggs/pull/2002/commits/5e4518ed0457cc698ca45fbec3881688a6d33a70 It wont work yet as the example config is not yet merged to the master branch.

If you upload your packages files and no config and start server I will generate whole config file for them and then they will just have to change what they need and restart.  It's all in my readme step by step

**Comment by QuintenQVD0** [comment]
> First startup will make a config for them to modify. If the parser is in it will clear it out and the server won't recover

that is wy there is now a example config when the server is installing. the just have to change 1 line and then the file parser won't clear it. I tryed it works

**Comment by gOOvER** [comment]
> > @kj4lxc @gOOvER I added back the file parser as the install script no download an example config. where they just have to change 1 line. Have a look at https://github.com/parkervcp/eggs/pull/2002/commits/5e4518ed0457cc698ca45fbec3881688a6d33a70 It wont work yet as the example config is not yet merged to the master branch.
> 
> If you upload your packages files and no config and start server It will generate whole config file for them and then they will just have to change what they need and restart.  It's all in my readme step by step
> 
> 

i think, the Parser create an empty File and the Server dont create/overwrite the existing File. 

i will do some Tests tomorrow with preshipped Files. The exported Files are only  needed, when you want use mods or Special DLC's

**Comment by QuintenQVD0** [comment]
> > > @kj4lxc @gOOvER I added back the file parser as the install script no download an example config. where they just have to change 1 line. Have a look at [5e4518e](https://github.com/parkervcp/eggs/commit/5e4518ed0457cc698ca45fbec3881688a6d33a70) It wont work yet as the example config is not yet merged to the master branch.
> > 
> > 
> > If you upload your packages files and no config and start server It will generate whole config file for them and then they will just have to change what they need and restart.  It's all in my readme step by step
> 
> i think, the Parser create an empty File and the Server dont create/overwrite the existing File.
> 
> i will do some Tests tomorrow with preshipped Files. The exported Files are only needed, when you want use mods or Special DLC's

ye it is the parser that creates a empty file and the server bin does not overwrite it, that is wy there now is a example file so the parser does not clear it

**Review by parkervcp — APPROVED** [review]
LGTM

**Comment by Dylix1** [comment]
It seems i solved the problem that the server isn't showing up in the Convoy browser, you need to asign the query_dedicated_port via the Assign Additional Ports section of the build configuration default it is port: 27016. 

it uses this port to communicate with the server browser

**Comment by gOOvER** [comment]
> It seems i solved the problem that the server isn't showing up in the Convoy browser, you need to asign the query_dedicated_port via the Assign Additional Ports section of the build configuration default it is port: 27016. 
> 
> it uses this port to communicate with the server browser
> 

did you Update from old egg? If yes, create a new Server 

The Egg works fine and Server is Shown in Serverbrowser

**Comment by HerrSammyDE** [comment]
Is this normal that my server does not take the configs from the repo? @QuintenQVD0 @gOOvER

Here is a screenshot:
chrome_dlgTtSuhQR

**Pull request merged**